draft-ietf-quic-http.md

```diff
@@ -585,6 +585,116 @@ The payload consists of:
   : HPACK-compressed request headers for the promised response.
 
 
+### GOAWAY {#frame-goaway}
+
+The GOAWAY frame (type=0x7) is used to initiate graceful shutdown of a
+connection.  GOAWAY allows an endpoint to stop accepting new streams while still
+finishing processing of previously established streams.  This enables
+administrative actions, like server maintenance.
+
+There is an inherent race condition between an endpoint starting new streams and
+the remote sending a GOAWAY frame.  To deal with this case, the GOAWAY contains
+the stream identifier of the last peer-initiated stream that was or might be
+processed on the sending endpoint in this connection.  For instance, if the
+server sends a GOAWAY frame, the identified stream is the highest-numbered
+stream initiated by the client.  This MAY be lower than the stream limit
+identified by a QUIC MAX_STREAM_ID frame.
+
+Once sent, the sender will refuse requests sent on streams initiated by the
+receiver if the stream has an identifier higher than the included last stream
+identifier.  Receivers of a GOAWAY frame MUST NOT send new requests on the
+connection, although a new connection can be established for new streams.
+
+If the receiver of the GOAWAY has sent requests on streams with a higher stream
+identifier than what is indicated in the GOAWAY frame, those streams are not or
+will not be processed.  The receiver of the GOAWAY frame can treat the requests
+as though they had never been sent at all, thereby allowing those requests to be
+retried later on a new connection.  Servers SHOULD treat requests on streams
+above this ID as stream errors of type HTTP_REQUEST_CANCELLED.
+
+Endpoints SHOULD send a GOAWAY frame when the closing of a connection is known
+in advance, even if the window is small, so that the remote peer can know
```

Comment:
If the server knows that it's going to close a connection, it should send GOAWAY to warn the client.  Errors that warrant CONNECTION_CLOSE aren't generally known in advance, but if they are, then I agree that this should include them.